design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+        exclude: [ string ]  # branches to discard the trigger events, optional; Defaults to none.
+      stages: [ string ]  # trigger after completion of given stage, optional; Defaults to all stage completion.
+      tags: [ string ]  # tags on which the trigger events are considered, optional; Defaults to any tag or no tag.
+```
+
+#### Examples
+Triggers are enabled by default unless expliciltly opted out. You can disable the triggers on the `pipeline` resource.
+```yaml
+resources:
+  pipelines:
+  - pipeline: SmartHotel
+    source: SmartHotel-CI 
+    trigger: none   
+```
+
+
+You can control which branches get the triggers with a simple syntax.
+```yaml
+resources:
+  pipelines:
+  - pipeline: SmartHotel
+    source: SmartHotel-CI 
+    trigger: 
+    - releases/*
+    - master
+ ```
+ You can specify the full name of the branch (for example, master) or a prefix-matching wildcard (for example, releases/*). You cannot put a wildcard in the middle of a value. For example, releases/*2018 is invalid.
+ 
+
+
+You can specify the branches to include and exclude.
+```yaml
+resources:
+  pipelines:
+  - pipeline: SmartHotel
+    source: SmartHotel-CI 
+    trigger: 
+      branches:
+        include: 
+        - releases/*
+        exclude:
+        - master
+ ```
+ 
+ 
+ You can specify which tags to control the triggers.
+ ```yaml
+resources:
+  pipelines:
+  - pipeline: SmartHotel
+    source: SmartHotel-CI 
+    trigger: 
+      branches:
+        include: 
+        - releases/*
+      tags: 
+      - Production
+      - Signed
+ ```
+ 
+ 
+If you don't want to wait until all the stages of the run are completed for the `pipeline` resource. You can provide the stage to be completed to trigger you pipeline. 
+```yaml
+resources:
+  pipelines:
+  - pipeline: SmartHotel
+    source: SmartHotel-CI 
+    trigger: 
+      branches:
+        include: master
+      stages: 
+      - QA
+ ```
+ 
+ 
+### Repositories
+Triggers can be set on `repository` resources defined the pipeline. See [repository resource](pipeline-resources.md#resources-repositories) for more details.
+
+For repositories, you can set two types of triggers.
+1. Repo triggers
+2. PR triggers
+
+
+#### Repo triggers
+Whenever a commit goes to your `repository`, a new pipeline run gets triggered. 
+
+
+#### Scenarios:	
+- I would like to trigger my pipeline only when a commit happens on ‘releases/*’ branch of the repository.
+- I would like to trigger my pipeline when a new commit happens, however, I would like to enable batching so that only one pipeline runs at a time. 
+- I would like to trigger my pipeline only when a new commit goes into the file path “Repository/Web/*”.
+
+
+#### Schema
+```yaml
+resources:       
+  repositories:
+  - repository: string    
+    type: enum 
+    connection: string 
+    source: string  
+    trigger:  # Optional; Triggers are enabled by default
```

Comment:
Making this enabled by default would be a breaking change. Also, for a non-`self` repository, I'm not sure we want to trigger by default. There are scenarios where the secondary repository shouldn't trigger (templates and external tools are two that come immediately to mind).

Reply:
I would imaging templates/tools to be the more common scenarios. I agree not trigger by default seems like the best behavior.

Reply:
As I have mentioned in the other conversation, for repos and containers, triggers will not be enabled by default in the current version of YAML.